clang/lib/AST/Type.cpp

```diff
@@ -5320,6 +5312,19 @@ LLVM_DUMP_METHOD void FunctionEffectSet::dump(llvm::raw_ostream &OS) const {
   FunctionEffectsRef(*this).dump(OS);
 }
 
+LLVM_DUMP_METHOD void FunctionEffectKindSet::dump(llvm::raw_ostream &OS) const {
+  OS << "Effects{";
```

Comment:
Replace this whole function body with:
```
OS << "Effects{";
llvm::interleaveComma(*this, OS);
OS << "}";
```

There is value, IMO to overriding operator `<<` for Effect to print its name, so this would require that.  Else, you can use a lambda for the `OS << Effect.name()` part.